pkg/fleet/installer/setup/djm/emr.go

```diff
@@ -0,0 +1,178 @@
+	ReleaseLabel string `json:"releaseLabel"`
+}
+
+// SetupEmr sets up the DJM environment on EMR
+func SetupEmr(s *common.Setup) error {
+
+	s.Packages.Install(common.DatadogAgentPackage, emrAgentVersion)
+	s.Packages.Install(common.DatadogAPMInjectPackage, emrInjectorVersion)
+	s.Packages.Install(common.DatadogAPMLibraryJavaPackage, emrJavaVersion)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("failed to get hostname: %w", err)
+	}
+	s.Config.DatadogYAML.Hostname = hostname
+	s.Config.DatadogYAML.DJM.Enabled = true
+
+	s.Config.DatadogYAML.ExpectedTagsDuration = "10m"
+	isMaster, clusterName, err := setupCommonEmrHostTags(s)
+	if err != nil {
+		return fmt.Errorf("failed to set tags: %w", err)
+	}
+	if isMaster == "true" {
+		setupEmrDriver(s, hostname, clusterName)
+	}
+	return nil
+}
+
+func setupCommonEmrHostTags(s *common.Setup) (string, string, error) {
+
+	instanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "instance.json"))
+	if err != nil {
+		return "", "", fmt.Errorf("error reading instance file: %w", err)
+	}
+
+	var info emrInstanceInfo
+	if err = json.Unmarshal(instanceInfoRaw, &info); err != nil {
+		return "", "", fmt.Errorf("error umarshalling instance file: %w", err)
+	}
+
+	setHostTag(s, "instance_group_id", info.InstanceGroupID)
+	setHostTag(s, "is_master_node", info.IsMaster)
+	s.Span.SetTag("host_tag."+"is_master_node", info.IsMaster)
+
+	extraInstanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "extraInstanceData.json"))
+	if err != nil {
+		return info.IsMaster, "", fmt.Errorf("error reading extra instance data file: %w", err)
+	}
+
+	var extraInfo extraEmrInstanceInfo
+	if err = json.Unmarshal(extraInstanceInfoRaw, &extraInfo); err != nil {
+		return info.IsMaster, "", fmt.Errorf("error umarshalling extra instance data file: %w", err)
+	}
+	setHostTag(s, "job_flow_id", extraInfo.JobFlowID)
+	setHostTag(s, "cluster_id", extraInfo.JobFlowID)
+	setHostTag(s, "cluster_id", extraInfo.JobFlowID)
+
+	emrResponseRaw, err := executeCommandWithTimeout("aws", "emr", "describe-cluster", "--cluster-id", extraInfo.JobFlowID)
+	var response EmrResponse
+	if err = json.Unmarshal(emrResponseRaw, &response); err != nil {
+		return info.IsMaster, "", fmt.Errorf("error unmarshalling AWS EMR response: %w", err)
+	}
+
+	setHostTag(s, "cluster_name", response.Cluster.Name)
+	return info.IsMaster, response.Cluster.Name, nil
+}
+
+func setupEmrDriver(s *common.Setup, host string, clusterName string) {
+
+	s.Packages.Install(common.DatadogAPMInjectPackage, emrInjectorVersion)
+	s.Packages.Install(common.DatadogAPMLibraryJavaPackage, emrJavaVersion)
+
+	s.Config.InjectTracerYAML.AdditionalEnvironmentVariables = tracerEnvConfigEmr
+
+	var sparkIntegration common.IntegrationConfig
+	var yarnIntegration common.IntegrationConfig
+
+	if host != "" {
+		sparkIntegration.Instances = []any{
+			common.IntegrationConfigInstanceSpark{
+				SparkURL:         "http://" + host + ":8088",
```

Comment:
shouldn't it be localhost?

Reply:
I'm referring to [this line](https://github.com/DataDog/dd-djm-init-scripts/blob/main/scripts/emr_init.sh.template#L139) from the current EMR init script, it doesn't seem to be localhost as HOST is set [here](https://github.com/DataDog/dd-djm-init-scripts/blob/main/scripts/emr_init.sh.template#L133)

Reply:
yes but host is the hostname in the script, so it's the localhost you should be able to use `127.0.0.1` straight and skip dns resolution